common/djangoapps/third_party_auth/middleware.py

```diff
@@ -23,3 +23,33 @@ def get_redirect_uri(self, request, exception):
             redirect_uri = pipeline.AUTH_DISPATCH_URLS[auth_entry]
 
         return redirect_uri
+
+
+class PipelineQuarantineMiddleware(object):
+    """
+    Middleware flushes the session if a user agent with a quarantined session
+    attempts to leave the quarantined set of views.
+    """
+
+    def process_view(self, request, view_func, view_args, view_kwargs):  # pylint: disable=unused-argument
+        """
+        Check the session to see if we've quarantined the user to a particular
+        step of the authentication pipeline; if so, look up which modules the
+        user is allowed to browse to without breaking the pipeline. If the view
+        that's been requested is outside those modules, then flush the session.
+
+        In general, this middleware should be used in cases where allowing the
+        user to exit the running pipeline would be undesirable, and where it'd
+        be better to flush the session state rather than allow it. This is
+        currently used in the Enterprise app to ensure retrieval of data sharing
+        consent for all users who sign in using a provider that requires it.
```

Comment:
For the last sentence, how about something like: "Pipeline quarantining is utilized by the Enterprise application to enforce collection of user consent for sharing data with a linked third-party authentication provider."